File: backend/src/main/java/com/team05/petmeeting/domain/naming/repository/NamingRepositoryCustom.java
package com.team05.petmeeting.domain.naming.repository;

import com.team05.petmeeting.domain.naming.dto.NameCandidateRes;
import org.jspecify.annotations.Nullable;

import java.util.List;

public interface NamingRepositoryCustom {

    NameCandidateRes getCandidates(
            Long animalId,
            @Nullable Long userId
    );

    List<NameCandidateRes.CandidateDto> findAllQualifiedCandidatesByShelter(String careNm, int threshold);

}

File: backend/src/main/java/com/team05/petmeeting/domain/naming/repository/NamingRepositoryCustom.kt
package com.team05.petmeeting.domain.naming.repository

import com.team05.petmeeting.domain.naming.dto.NameCandidateRes
import com.team05.petmeeting.domain.naming.dto.NameCandidateRes.CandidateDto

interface NamingRepositoryCustom {
    fun getCandidates(
        animalId: Long,
        userId: Long?
    ): NameCandidateRes

    fun findAllQualifiedCandidatesByShelter(careNm: String, threshold: Int): List<CandidateDto>
}
